cpp/include/raft/random/rmat_rectangular_generator.cuh

```diff
@@ -84,6 +139,38 @@ void rmat_rectangular_gen(IdxT* out,
  * distributions across all the scales
  */
 template <typename IdxT, typename ProbT>
+void rmat_rectangular_gen(const raft::handle_t& handle,
+                          raft::random::RngState& r,
+                          raft::device_vector_view<IdxT, IdxT> out,
+                          raft::device_vector_view<IdxT, IdxT> out_src,
```

Comment:
Same here- we should be able to make these optional.

Reply:
I've consolidated the output vectors into a class that makes the meaning of the output vectors more clear, validates sizes, and prevents users from getting into an unsupported or incorrect state.  Thanks!